packages/react/menu/src/Menu.tsx

```diff
@@ -62,9 +63,10 @@ Menu.displayName = MENU_NAME;
 
 const CONTENT_NAME = 'MenuContent';
 
+type MenuItemRef = React.RefObject<React.ElementRef<typeof MenuItem>>;
 type MenuContentContextValue = {
-  contentRef: React.RefObject<HTMLDivElement>;
-  onItemsReachableChange(open: boolean): void;
+  itemMap: Map<MenuItemRef, MenuItemRef>;
```

Comment:
I haven't no, it's just something that came to my mind when I was trying out that sandbox. I'll dig.

Reply:
Heads up, after checking this intuition, it turned out to be true, and if the users memoizes an item, it absolutely destroys our solution and makes the order unpredictable… So we're back to the drawing board on the tracking aspect. After discussing with @jjenzz, we think we're going to go back to the DOM again but perhaps in something a bit more of a general solution as it's needed in a few places, and we realised that other components are broken because of the same issue.